container/kvm/cache_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package kvm_test
+
+import (
+	"net/http"
+
+	gc "gopkg.in/check.v1"
+
+	. "github.com/juju/juju/container/kvm"
+	"github.com/juju/juju/environs/imagedownloads"
+	jc "github.com/juju/testing/checkers"
+	"github.com/pkg/errors"
+)
+
+// cacheSuite is gocheck boilerplate.
+type cacheSuite struct{}
+
+// _ is gocheck boilerplate.
+var _ = gc.Suite(cacheSuite{})
+
+func (cacheSuite) TestSyncOnerErrors(c *gc.C) {
+	o := fakeParams{FakeData: nil, Err: errors.New("oner failed")}
```

Comment:
I feel like these lines could be reduced if you had a test variables and setup. The tests themselves would almost become one liners \o/